scripts/stablex/transitive_orderbook.js

```diff
@@ -0,0 +1,115 @@
+const BatchExchangeViewer = artifacts.require("BatchExchangeViewer")
+const { decodeOrdersBN } = require("../../src/encoding.js")
+const BN = require("bn.js")
+
+const { Orderbook, Offer } = require("../../typescript/common/orderbook.js")
+const { Fraction } = require("../../typescript/common/fraction.js")
+
+const argv = require("yargs")
+  .option("sellToken", {
+    describe: "the token you are looking to sell",
+  })
+  .option("buyToken", {
+    describe: "the token you are looking to sell",
+  })
+  .option("sellAmount", {
+    describe: "the amount you are looking to sell",
+  })
+  .option("pageSize", {
+    default: 100,
+    describe: "The page  size for the function getOrdersPaginated",
+  })
+  .demand(["sellToken", "buyToken", "sellAmount"])
+  .version(false).argv
+
+const addItemToOrderbooks = function(orderbooks, item) {
+  let orderbook = new Orderbook(item.sellToken, item.buyToken)
+  if (!orderbooks.has(orderbook.pair())) {
+    orderbooks.set(orderbook.pair(), orderbook)
+  }
+  orderbook = orderbooks.get(orderbook.pair())
+  const price = new Fraction(item.priceNumerator, item.priceDenominator)
+  const volume = new Fraction(item.remainingAmount.gt(item.sellTokenBalance) ? item.sellTokenBalance : item.remainingAmount, 1)
+  // Smaller orders cannot be matched anyways
+  if (volume.gt(new Fraction(10000, 1))) {
+    orderbook.addAsk(new Offer(price, volume))
+  }
+}
+
+const getAllOrderbooks = async function(instance, pageSize) {
```

Comment:
This function is quite similar to https://github.com/gnosis/dex-contracts/blob/master/scripts/stablex/utilities.js#L102

But since it is not the same, it would be good to have a general function in utitilies: `getOpenOrdersPaginated`, which basically contains your logic, but not the orderbooks conversion.

If you have that then you can rewrite your function to:
```
const getAllOrderbooks = async function(instance, pageSize) {
       // getting orders:
     orders = getOpenOrdersPaginated(instance, pageSize)
     const orderbooks = new Map()
     elements.forEach(item => {
         addItemToOrderbooks(orderbooks, item)
    })
     return orderbooks
}
```
and we gain another reusable function.

Reply:
Good though!

One difference is that `getOrdersPaginated` works on `BatchExchange` whereas this one works on `BatchExchangeViewer`. Without strong types this might be somewhat confusing, but I'm happy to make this function reusable.

I'm hoping we can move this codebase over to type script in the near future.

Reply:
Yes, I think we should stop calling "instance" to the contracts, and use the contract names. I find it confusing. 
I would also be in favour of moving this logic also out of the scripts and decoupled it from the "orderbooks" part